crypto/x509/x509_vfy.c

```diff
@@ -568,8 +568,9 @@ static int check_chain_extensions(X509_STORE_CTX *ctx) {
   int ok = 0, plen = 0;
   int purpose = ctx->param->purpose;
 
-  // Check all untrusted certificates
-  for (int i = 0; i < ctx->last_untrusted; i++) {
+  int num = (int)sk_X509_num(ctx->chain);
+
+  for (int i = 0; i < num; i++) {
```

Comment:
Just to verify, this is essentially making our certificate checks slightly stricter but aligning with 1.1.1?

Reply:
Correct you can see the equivalent line here: https://github.com/openssl/openssl/blob/OpenSSL_1_1_1-stable/crypto/x509/x509_vfy.c#L485 where `num` is set to `sk_X509_num(ctx->chain)` on line 462.